tests/kernel/mem_protect/mem_protect/src/mem_domain.c

```diff
@@ -91,7 +91,7 @@ static inline void set_valid_fault_value(int test_case_number)
 /* Userspace function */
 void mem_domain_for_user(void *tc_number, void *p2, void *p3)
 {
-	set_valid_fault_value((u32_t)tc_number);
+	set_valid_fault_value((uintptr_t)tc_number);
```

Comment:
`set_valid_fault_name` currently accepts `int`. This will spit out a truncation warning in 64-bit environments where `int` is defined as a 32-bit type.

Reply:
AFAICT this currently doesn't generate a warning in such environments. Is there some non-default -W setting which shows this?

Reply:
> AFAICT this currently doesn't generate a warning in such environments. Is there some non-default -W setting which shows this?

Hmm, this is rather interesting. Microsoft compilers, by default, display a warning when implicitly converting from a 64-bit to a 32-bit type, but that is not the case for gcc and clang. Apparently, even `-Wall` does not enable this and you have to use `-Wconversion`:

```
stephanos@DOMINUS:~/test$ /opt/sdk/zephyr-sdk-0.11.0-alpha-7/aarch64-zephyr-elf/bin/aarch64-zephyr-elf-gcc -nostdlib -o test2 test2.c -Wconversion
test2.c: In function 'main':
test2.c:11:11: warning: conversion from 'uintptr_t' {aka 'long unsigned int'} to 'int' may change value [-Wconversion]
   11 |  deadbeef(i_am_64_bit_type);
      |           ^~~~~~~~~~~~~~~~
/opt/sdk/zephyr-sdk-0.11.0-alpha-7/aarch64-zephyr-elf/bin/../lib/gcc/aarch64-zephyr-elf/9.2.0/../../../../aarch64-zephyr-elf/bin/ld: warning: cannot find entry symbol _start; defaulting to 0000000000400000
```

Either way, it is true that there is an __implicit__ type down-conversion/truncation if we do `set_valid_fault_value(uintptr_t_value)` and the best practice would be to explicitly cast `tc_number` to `int` here to remove any ambiguity.

Reply:
no complaints here -- I'll revise this.